app/javascript/mastodon/actions/statuses.js

```diff
@@ -92,12 +92,17 @@ export function fetchStatus(id) {
 
     dispatch(fetchStatusRequest(id, skipLoading));
 
-    asyncDB.then(db => {
+    openDB().then(db => {
       const transaction = db.transaction(['accounts', 'statuses'], 'read');
       const accountIndex = transaction.objectStore('accounts').index('id');
       const index = transaction.objectStore('statuses').index('id');
 
-      return getFromDB(dispatch, getState, accountIndex, index, id);
+      return getFromDB(dispatch, getState, accountIndex, index, id).then(() => {
+        db.close();
+      }, error => {
```

Comment:
Is the `then(callback, errorCallback)` syntax preferred over `then(callback).catch(callback)`?

Reply:
No. It is a semantic pattern rather than a syntax. We see `then(onresolution).catch(onrejection)` even more often than `then(onresolution, onrejection)` because the former can catch errors in `onresolution` while the latter cannot.

In this case, however, the two callbacks do the same thing and catching an error of a callback with the other one does not make sense.